core/src/impl/Kokkos_Profiling_Interface.cpp

```diff
@@ -52,6 +52,20 @@ namespace Kokkos {
        	return (NULL != initProfileLibrary);
     }
 
+    void pushRegion(const std::string& kName) {
+	if( NULL != pushRegionCallee ) {
+	    Kokkos::fence();
```

Comment:
Are push & pop regions relative to a particular execution space instance, or are they global?
If global then how will concurrent push & pop from different execution space instances be handled?